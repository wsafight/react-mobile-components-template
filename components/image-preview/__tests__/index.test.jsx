import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import ImagePreview from '../index';

const images = [
  'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  'https://static.zhongan.com/website/health/zarm/images/banners/2.png',
  'https://static.zhongan.com/website/health/zarm/images/banners/3.png',
];

const originImages = [
  {
    url: 'https://cdn-health.zhongan.com/zarm/imagePreview/compress_1.png',
    originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  },
  {
    url: 'https://cdn-health.zhongan.com/zarm/imagePreview/compress_2.png',
    originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/2.png',
  },
  {
    url: 'https://cdn-health.zhongan.com/zarm/imagePreview/compress_3.png',
    originUrl: 'https://static.zhongan.com/website/health/zarm/images/banners/3.png',
  },
];

describe('ImagePreview', () => {
  it('renders correctly', () => {
    const wrapper = render(<ImagePreview checked onChange={jest.fn()} images={images} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with origin', () => {
    const wrapper = render(<ImagePreview checked onChange={jest.fn()} images={originImages} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
