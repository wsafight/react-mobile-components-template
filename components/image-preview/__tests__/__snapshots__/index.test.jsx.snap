// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview renders correctly 1`] = `null`;

exports[`ImagePreview renders correctly with origin 1`] = `null`;
